test/e2e/common.go

```diff
@@ -175,3 +177,99 @@ func deploymentRolledOut(ctx context.Context, clientSet *kubernetes.Clientset, n
+	bmhs := bmov1alpha1.BareMetalHostList{}
+	Expect(c.List(ctx, &bmhs, opts...)).To(Succeed())
+	Logf("Listing BareMetalHosts:")
+	Logf("Name	Status	Consumer	Online")
+	Logf("---------------------------------------------------------------------------------")
+	for _, bmh := range bmhs.Items {
+		consumer := ""
+		if bmh.Spec.ConsumerRef != nil {
+			consumer = bmh.Spec.ConsumerRef.Name
+		}
+		Logf("%s	%s	%s	%t", bmh.GetName(), bmh.Status.Provisioning.State, consumer, bmh.Status.PoweredOn)
+	}
+	Logf("---------------------------------------------------------------------------------")
+	Logf("%d BareMetalHosts in total", len(bmhs.Items))
+	Logf("=================================================================================")
+}
+
+// listMetal3Machines logs the names, ready status and provider ID of all Metal3Machines in the namespace.
+// Similar to kubectl get metal3machines.
+func listMetal3Machines(ctx context.Context, c client.Client, opts ...client.ListOption) {
+	metal3Machines := infrav1.Metal3MachineList{}
+	Expect(c.List(ctx, &metal3Machines, opts...)).To(Succeed())
+	Logf("Listing Metal3Machines:")
+	Logf("Name	Ready	Provider ID")
+	Logf("---------------------------------------------------------------------------------")
+	for _, metal3Machine := range metal3Machines.Items {
+		providerID := ""
+		if metal3Machine.Spec.ProviderID != nil {
+			providerID = *metal3Machine.Spec.ProviderID
+		}
+		Logf("%s	%t	%s", metal3Machine.GetName(), metal3Machine.Status.Ready, providerID)
+	}
+	Logf("---------------------------------------------------------------------------------")
+	Logf("%d Metal3Machines in total", len(metal3Machines.Items))
+	Logf("=================================================================================")
+}
+
+// listMachines logs the names, status phase, provider ID and Kubernetes version
+// of all Machines in the namespace. Similar to kubectl get machines.
+func listMachines(ctx context.Context, c client.Client, opts ...client.ListOption) {
+	machines := clusterv1.MachineList{}
+	Expect(c.List(ctx, &machines, opts...)).To(Succeed())
+	Logf("Listing Machines:")
+	Logf("Name	Status	Provider ID	Version")
+	Logf("---------------------------------------------------------------------------------")
+	for _, machine := range machines.Items {
+		providerID := ""
+		if machine.Spec.ProviderID != nil {
+			providerID = *machine.Spec.ProviderID
+		}
+		Logf("%s	%s	%s	%s", machine.GetName(), machine.Status.GetTypedPhase(), providerID, *machine.Spec.Version)
+	}
+	Logf("---------------------------------------------------------------------------------")
+	Logf("%d Machines in total", len(machines.Items))
+	Logf("=================================================================================")
+}
+
+// listNodes logs the names, status and Kubernetes version of all Nodes.
+// Similar to kubectl get nodes.
+func listNodes(ctx context.Context, c client.Client) {
```

Comment:
Any reasons this doesn't include opts?

Reply:
Just no need for now. All the other resources we list are namespaced, so we need to at least say what namespace to look in. Nodes are cluster wide so no need. We could at some point add filters for control plane and workers but for now we don't use it for anything like that